src/index.js

```diff
@@ -55,3 +55,54 @@ userTask.addEventListener('keyup', (e) => {
 });
 
 displayTask(tasksArray);
+
+const updateIndex = () => {
+  for (let i = 0; i < tasksArray.length; i += 1) {
+    tasksArray[i].id = i + 1;
+  }
+  tasksArray.sort((a, b) => {
+    if (a.id < b.id) {
+      return -1;
+    }
+    if (a.id > b.id) {
+      return 1;
+    }
+    return 0;
+  });
+};
+
+taskSection.addEventListener('click', (e) => {
+  if (e.target.classList.contains('span')) {
+    tasksArray.splice(e.target.parentElement, 1);
+    updateIndex();
+    displayTask(tasksArray);
+    localStorage.setItem('task', JSON.stringify(tasksArray));
+  }
+
+  if (e.target.classList.contains('checkBoxClass')) {
+    if (e.target.checked) {
+      tasksArray[e.target.id].status = true;
+      localStorage.setItem('task', JSON.stringify(tasksArray));
+      e.target.parentElement.classList.add('checked');
+    } else {
+      tasksArray[e.target.id].status = false;
+      localStorage.setItem('task', JSON.stringify(tasksArray));
+      displayTask(tasksArray);
+      e.target.parentElement.classList.remove('checked');
+    }
```

Comment:
- [ ] In the project requirements, we are encouraged to organize our codebase into modules(files). This is necessary for having a clean codebase that is easy to follow and maintain.

  We, therefore, need to kindly refactor this part of the code to satisfy the requirement quoted below:
  
  > - Add a new JavaScript file and import it as a module:
  >    - it will contain methods related to the status updates (completed: true/false).
  
- [ ] The other requirement is adding the event listener `change`. This hasn't been implemented even though the code as implemented works. I am hoping we could both realize it is a lot of code in the implementation and that using the `change` event listener would clean our code a bit. Please see below:

  Add another event listener(change listener for the textboxes) to taskSection. Don't worry about it's acceptable 👇🏼:
  
  ```JavaScript
  taskSection.addEventListener('change', (event) => {
    if (event.target.className === 'checkBoxClass') {
      tasksArray[event.target.id].status = !tasksArray[event.target.id].status;
      localStorage.setItem('task', JSON.stringify(tasksArray));
      event.target.parentElement.classList.toggle('checked');
    }
  });
  ```  
  From the above implementation, do you realize our lines of code have been reduced by half?
  
  When you implement this, we'll get to satisfy the requirement quoted below:
  
  > - Add event listener to the checkbox (change).